build_docs

```diff
@@ -169,7 +169,13 @@ def run_build_docs(args):
             extra_name = repo_name.replace('x-pack-', '')
             repo_mount = '/doc/%s-extra/%s' % (extra_name, repo_name)
         else:
-            repo_mount = '/doc/' + repo_name
+            match = re.search(r'(.*)-[0-9]+\.[0-9x]+$', repo_name)
```

Comment:
This seems like a pretty safe way to detect "repo names" that include a `-<version branch>` suffix, and is unlikely to inadvertently catch a non-worktree repo name; the only repo that contains a digit right now is `cloud-on-k8s`.

Is this pattern for worktree names (`elasticsearch`, `elasticsearch-7.x`, `elasticsearch-6.x`) pretty universal and documented anywhere? I wouldn't want to have a partial solution that only works for this specific setup if it's not universal; if I understand correctly, any other worktrees that don't match this pattern would still fail, right?

Would it be possible to parse the `.git` file (which on my machine seems to contain something like `gitdir: /absolute/path/to/elasticsearch/.git/worktrees/elasticsearch-7.x` and find the repo name by looking at the path component before `.git`?

Reply:
I don't think this would match the pattern I am using for my work trees. I am not aware of any conventions/documentation for how to create these (I have a tiny bash script for personal use)
```
~/workspace/elasticsearch $ ls
7.0  7.1  7.10  7.11  7.12  7.13  7.14  7.15  7.16  7.17  7.2  7.3  7.4  7.5  7.6  7.7  7.8  7.9  8.0  8.1  8.2  main
```

Reply:
This is the format that we recommend in the ES development docs
```
$ git fetch origin 7.x
$ git checkout -b 7.x origin/7.x && git checkout master
$ git worktree add ../elasticsearch-7.x 7.x
```

I'll try and pick this PR up again and see if I can make it agnostic to the worktree name.

Reply:
Ahh.. i had missed those.  Not a big fan of that layout and the instructions needed updating anyway ... so I am taking a swing at changing the recommendation : https://github.com/elastic/elasticsearch-team/pull/882